### PR TITLE
fix(forms): Update validation forms

### PR DESCRIPTION
**Pull request by jefflembeck**
This fix includes:
- If a form is not valid, the submit button is disabled
- Refactoring to OOJS for easier testing

### CODE REVIEW

File: src/pivotal-ui/components/forms/validator.js [hunk 1]
```diff
@@ -61,5 +38,49 @@ var handleInput = function handleInput(e) {
...
+    this.submitButton.attr("disabled", "");
+  }
+};
+
+$(function(){
+  var forms = $("form[data-validate]");
+
+  $.each(forms, function(idx, form){
+    var vf = new ValidatedForm(form);
+    vf.addListeners();
```
Comment (aredridel) [inline src/pivotal-ui/components/forms/validator.js:83]: Why do you make this a separate method?

File: src/pivotal-ui/components/forms/validator.js [hunk 2]
```diff
@@ -61,5 +38,49 @@ var handleInput = function handleInput(e) {
...
+
+  if(!this.form[0].checkValidity()){
+    this.submitButton.attr("disabled", "");
+  }
 };
+
+ValidatedForm.prototype.addListeners = function addListners() {
+  this.form.on("submit", handleSubmit.bind(this));
+  this.form.on("input", handleInput.bind(this));
+  this.form.on("click", "input[type='checkbox'], input[type='radio']", handleInput.bind(this));
```
Comment (aredridel) [inline src/pivotal-ui/components/forms/validator.js]: click, not changed? What about keyboard interaction?

Comment (aredridel) [inline src/pivotal-ui/components/forms/validator.js]: Also this is totally not in the commit message.

**Comment by aredridel** [comment]
Looks good to me, comments aside

File: src/pivotal-ui/components/forms/validator.js (hunk 2 above)
Comment (jefflembeck) [inline src/pivotal-ui/components/forms/validator.js]: Doh, forgot `change`

File: src/pivotal-ui/components/forms/validator.js (hunk 1 above)
Comment (jefflembeck) [inline src/pivotal-ui/components/forms/validator.js:83]: I'd like to separate all the concerns for testing so  +/- every line of the constructor method is testable.

Comment (aredridel) [inline src/pivotal-ui/components/forms/validator.js:83]: Aah alright!

**Pull request merged**